### DISCUSSION TITLE
How to use heex and LiveView Components?

### DISCUSSION THREAD

**Discussion by cvkmohan** [mhanberg/temple#155] [Q&A] [answered]
Not sure I am framing the question right. 
But, the new heex syntax is it supported via temple? 
```<.form></.form>``` is the syntax for LiveView forms. Further Phoenix Component have a syntax like ```<.Button></.Button>``` How do we represent them in temple?
If they are already supported, a few examples in README will help immensely.

**Comment by mhanberg** [comment]
HEEx is completely unrelated to Temple. 

If you want to use components, you would use a Temple.Component. 

Temple also supports HTML aware things like Boolean attributes, but once again that is completely unrelated to HEEx.

**Comment by cvkmohan** [comment]
Thanks for the quick response. I think I understood the answer - atleast partially.
I would still think - it will be helpful for newbies - if there are some examples on how to use Phoenix.Components - and - LiveView Forms with events using temple templating language.

**Comment by mhanberg** [comment]
So you don't need to use `Phoenix.Component` at all, you can just use `Temple.Component`.

With regard to live view forms, i believe that the builtin `<.form></.form>` component can be recreated as a temple component.

I've tried to not bake in too much phoenix and live view specific things into Temple itself, as Temple should be able to be used without plug, phx, etc.

I can try to get a copy pastable form component that mimics the new `<.form>` phx component written up.

**Comment by mhanberg** [comment, marked as answer]
You can see what I do currently [here](https://github.com/mhanberg/temple/discussions/113#discussioncomment-482081).

I haven't worked on my "dogfooding" app in a bit so it's still on live view 0.15

**Reply by cvkmohan** [reply]
Thanks @mhanberg - It is clear that I understood the role and responsibilities of Temple wrongly. Your response and pointers cleared some of them. I will continue exploring.
